src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -197,9 +210,24 @@ internal void AddProvider(string workerDir)
             }
         }
 
-        private List<WorkerDescriptionProfile> ReadWorkerDescriptionProfiles(JToken profilesJToken)
+        private static JsonElement GetWorkerConfigJsonElement(string workerConfigPath)
         {
-            var profiles = profilesJToken.ToObject<IList<WorkerProfileDescriptor>>();
+            ReadOnlySpan<byte> jsonSpan = File.ReadAllBytes(workerConfigPath).AsSpan();
```

Comment:
nit: should null check workerConfigPath at start

Reply:
The caller already checks the file path is valid before calling this private method. So should be good.